lib/matplotlib/tests/test_dates.py

```diff
@@ -511,6 +511,113 @@ def _create_auto_date_locator(date1, date2):
         assert strings == expected
 
 
+def test_concise_formatter_formats():
+    formats = ['%Y', '%m/%Y', 'day: %d',
+               '%H hr %M min', '%H hr %M min', '%S.%f sec']
+
+    def _create_auto_date_locator(date1, date2):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(locator, formats=formats)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = []
+        for st in ax.get_yticklabels():
+            sts += [st.get_text()]
+        return sts
+
+    d1 = datetime.datetime(1997, 1, 1)
+    results = ([datetime.timedelta(weeks=52 * 200),
```

Comment:
The indentation here is strange. I'm suprised it's not being picked up by the pep8 checker. Can you check this is pep8 compatible?